nselib/tls.lua

```diff
@@ -1319,16 +1319,45 @@ end
 -- @param t Table of options
 -- @return The client_hello record as a string
 function client_hello(t)
-  local b, ciphers, compressor, compressors, h, len
+  local b, ciphers, compressor, compressors, h, len, protocol
   t = t or {}
 
   ----------
   -- Body --
   ----------
 
+  --print_r(t.ciphers)
   b = {}
   -- Set the protocol.
-  local protocol = t["protocol"] or HIGHEST_PROTOCOL
+  if not t.ciphers then
+    protocol = t["protocol"] or HIGHEST_PROTOCOL
+  -- #t.ciphers will be zero when we are sending many protocols
+  elseif #t.ciphers == 0 then
+    if t['protocol'] then
+      protocol = t['protocol']
+    else
+      if t.ciphers['TLSv1.2'] then
```

Comment:
@dmiller-nmap  I've checked here for all `PROTOCOLS` manually because using `PROTOCOLS` was making this piece of code too complex.